app/Http/Controllers/OrderHistoryController.php

```
@@ -1,48 +1,70 @@
-        return response()->json(OrderHistoryResource::collection($orderHistories));
+        // Build normalized filters and pagination signature for cache key
+        $filters = [
+            'order_id' => $request->input('order_id'),
+            'from_date' => $request->input('from_date'),
+            'to_date' => $request->input('to_date'),
+            'page' => (int) $request->input('page', 1),
+            'per_page' => (int) $request->input('per_page', 15),
+        ];
+
+        $key = self::indexKeyFor($filters);
+        $hit = Cache::has($key);
```

PHPMD (2.15.0)

Avoid using static access to class '\Illuminate\Support\Facades\Cache' in method 'index'. (undefined)

(StaticAccess)